### PR TITLE
feat(CardGroup): new web component

### PR DESCRIPTION
**Pull request by ariellalgilmore**
### Related Ticket(s)

#3552 

### Description

Create cardgroup for web component

### Changelog

**New**

- add stories, tests, and web component

### CODE REVIEW

File: packages/web-components/src/components/card-group/__stories__/card-group.stories.ts [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+    </dds-card-footer>
+  </dds-card-group-item>
+`;
+
+const cardGroupItemWithImages = html`
+  <dds-card-group-item href="https://example.com">
+    <dds-image slot="image" alt="Image alt text" default-src="https://dummyimage.com/1056x792/ee5396/161616&text=4:3">
+    </dds-image>
+    <slot slot="eyebrow">Topic</slot>
+    <slot slot="heading">Natural Language Processing.</slot>
```
Comment (asudoh) [inline packages/web-components/src/components/card-group/__stories__/card-group.stories.ts:80]: `<slot>` defines an "insertion point" in shadow DOM. For using a slot, using `slot` attribute is enough.

```suggestion
    <div slot="eyebrow">Topic</div>
    <div slot="heading">Natural Language Processing.</div>
```

File: packages/web-components/src/components/card-group/__stories__/card-group.stories.ts [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+  `;
+};
+
+export const withImagesAndCTA = ({ parameters }) => {
+  const { cards } = parameters?.props?.CardGroup ?? {};
+  return html`
+    <dds-card-group>
+      ${cards}
+      <dds-card-group-item href="https://example.com" color-scheme="inverse">
+        <slot slot="heading">Top level card link</slot>
```
Comment (asudoh) [inline packages/web-components/src/components/card-group/__stories__/card-group.stories.ts:53]: `<slot>` defines an "insertion point" in shadow DOM. For using a slot, using `slot` attribute is enough.

```suggestion
        <div slot="heading">Top level card link</div>
```

**Review by asudoh — COMMENTED** [review]
Nice progress @ariellalgilmore!

**Review by asudoh — APPROVED** [review]
LGTM 👍 - Thanks @ariellalgilmore!

**Comment by jeffchew** [comment]
Hi @ariellalgilmore , looks good! I see it's still in draft, so I guess you're still doing finishing touches (e.g. README)?

File: packages/web-components/tests/snapshots/dds-card-group.md [hunk 3]
```diff
@@ -0,0 +1,34 @@
+# `dds-card-group`
+
+#### `Renders Default`
+
+```
+<slot>
```
Comment (jeffchew) [inline packages/web-components/tests/snapshots/dds-card-group.md:6]: @ariellalgilmore why is the snapshot blank?

Comment (ariellalgilmore) [reply packages/web-components/tests/snapshots/dds-card-group.md:6]: Thanks @jeffchew!! Will update!

**Review by jeffchew — CHANGES_REQUESTED** [review]
@ariellalgilmore can you add a README?

**Comment by asudoh** [comment]
@jeffchew FYI I think @ariellalgilmore has addressed your review comment.

**Review by jeffchew — APPROVED** [review]
Thank you @ariellalgilmore , approved!

**Pull request merged**